.ci/scripts/prepare_slack_data.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python
+"""
+This script is designed to be used in a GitHub Actions workflow to send Slack notifications.
+It reads environment variables set by the GitHub Actions runtime, validates them, and
+constructs a Slack payload based on the workflow status.
+"""
+import os
+import json
+
+
+def check_env_var(env_var: str) -> str:
+    """
+    Retrieve the value of the specified environment variable.
+
+    Parameters:
+        env_var (str): The name of the environment variable to retrieve.
+
+    Returns:
+        str: The value of the specified environment variable.
+    """
+    value = os.environ.get(env_var)
+    if value is None or value == "":
+        raise ValueError(f"The env var '{env_var}' isn't defined.")
+    return value
+
+
+def set_output(name: str, value: str):
+    """
+    Set an output variable for the GitHub Actions workflow.
+
+    Parameters:
+        name (str): The name of the output variable.
+        value (str): The value to set for the output variable.
+    """
+    with open(os.environ["GITHUB_OUTPUT"], "a", encoding="utf-8") as fh:
+        print(f"{name}={value}", file=fh)
+
+
+def set_failed(message: str):
+    """
+    Set the GitHub Actions workflow status to 'failed' with a specified message.
+
+    Parameters:
+        message (str): The message to be associated with the failure.
+    """
+    print(f"::set-failed::{message}")
+
+
+def color_by_job_status(status: str) -> str:
+    """
+    Determine the Slack color based on the GitHub Actions job status.
+
+    Parameters:
+        status (str): The GitHub Actions job status, e.g., "success", "failure", etc.
+
+    Returns:
+        str: The Slack color corresponding to the GitHub Actions job status.
+             Possible values: "good" for success, "danger" for failure, or an empty string.
+    """
+    if status == "success":
+        return "#36a64f"
+    if status == "failure":
+        return "#D40E0D"
+    return ""
+
+
+class BuildSlackException(Exception):
+    """
+    Custom exception class for errors related to building Slack notifications.
+    """
```

Comment:
Why do we need an empty wrapper exception? who's going to raise it?

Reply:
The `run()` function, which executes the main flow, contains a try-except block. Providing a general Exception to catch is flagged as a warning by `pylint`. To address this, a custom BuildSlackException class was created to serve as an empty wrapper exception.

Reply:
Removed empty wrapper and improved code for better clarity and error handling.